app/src/main/java/ai/elimu/analytics/util/SharedPreferencesHelper.kt

```
@@ -0,0 +1,52 @@
+
+    @JvmStatic
+    fun getLanguage(context: Context): Language? {
+        Timber.i("getLanguage")
+        val sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
+        val languageAsString = sharedPreferences.getString(PREF_LANGUAGE, null)
+        return if (TextUtils.isEmpty(languageAsString)) {
+            null
+        } else {
+            try {
+                Language.valueOf(languageAsString!!)
+            } catch (e: IllegalArgumentException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)